gunrock/app/kcore/kcore_app.cu

```diff
@@ -0,0 +1,321 @@
+  bool quiet_mode = parameters.Get<bool>("quiet");
+  int num_runs = parameters.Get<int>("num-runs");
+  std::string validation = parameters.Get<std::string>("validation");
+  util::Info info("kcore", parameters, graph);
+
+  util::CpuTimer cpu_timer, total_timer;
+  cpu_timer.Start();
+  total_timer.Start();
+
+  // <TODO> get problem specific inputs, e.g.:
+  // std::vector<VertexT> srcs = parameters.Get<std::vector<VertexT>>("srcs");
+  // printf("RunTests: %d srcs: src[0]=%d\n", srcs.size(), srcs[0]);
+  // </TODO>
+
+  // <TODO> allocate problem specific host data, e.g.:
+  SizeT *h_k_cores = new SizeT[graph.nodes];
+  // </TODO>
+
+  // Allocate problem and enactor on GPU, and initialize them
+  ProblemT problem(parameters);
+  EnactorT enactor;
+  GUARD_CU(problem.Init(graph, target));
+  GUARD_CU(enactor.Init(problem, target));
+
+  cpu_timer.Stop();
+  parameters.Set("preprocess-time", cpu_timer.ElapsedMillis());
+
+  for (int run_num = 0; run_num < num_runs; ++run_num) {
+    GUARD_CU(problem.Reset(
+        // <TODO> problem specific data if necessary, eg:
+        // src,
+        graph,
+        // </TODO>
+        target));
+    GUARD_CU(enactor.Reset(
+        // <TODO> problem specific data if necessary:
+        // srcs[run_num % srcs.size()],
+        // </TODO>
+        target));
+
+//    util::PrintMsg("__________________________", !quiet_mode);
+    util::PrintMsg("__________________________", true);
+
+    cpu_timer.Start();
+    GUARD_CU(enactor.Enact(
+        // <TODO> problem specific data if necessary:
+        // srcs[run_num % srcs.size()]
+        // </TODO>
+        ));
+    cpu_timer.Stop();
+    info.CollectSingleRun(cpu_timer.ElapsedMillis());
+
+    /*util::PrintMsg(
+        "--------------------------\nRun " + std::to_string(run_num) +
+            " elapsed: " + std::to_string(cpu_timer.ElapsedMillis()) +
+            ", #iterations = " +
+            std::to_string(enactor.enactor_slices[0].enactor_stats.iteration),
+        !quiet_mode);*/
+    util::PrintMsg(
+        "--------------------------\nRun " + std::to_string(run_num) +
+            " elapsed: " + std::to_string(cpu_timer.ElapsedMillis()) +
+            ", #iterations = " +
+            std::to_string(enactor.enactor_slices[0].enactor_stats.iteration),
+        true);
+
+    if (validation == "each") {
+      GUARD_CU(problem.Extract(h_k_cores));
+      SizeT num_errors = Validate_Results(parameters, graph, h_k_cores, ref_k_cores, false);
+    }
+  }
+
+  cpu_timer.Start();
+
+  GUARD_CU(problem.Extract(h_k_cores));
+  if (validation == "last") {
+    SizeT num_errors = Validate_Results(parameters, graph, h_k_cores, ref_k_cores, false);
+  }
+
+  // compute running statistics
+  // <TODO> change NULL to problem specific per-vertex visited marker, e.g.
+  // h_distances
+  info.ComputeTraversalStats(enactor, (VertexT *)NULL);
+// Display_Memory_Usage(problem);
+#ifdef ENABLE_PERFORMANCE_PROFILING
+  // Display_Performance_Profiling(&enactor);
```

Comment:
Should this be commented?

Reply:
I have no idea. It seems like this was commented out in `hello_app.cu`, and that's what I used that as the template for this app.